File: src/main/java/de/blankedv/sx4draw/GenericAddress.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package de.blankedv.sx4draw;

import static de.blankedv.sx4draw.Constants.LBMIN;

/**
 * @author mblank
 */
public class GenericAddress {
    public int addr = 0;
    public int inv = 0;
    public int orient = 0;  // orientation of signals (0 = 0 grad, 1 = 90 etc)

    GenericAddress() {
    }


    GenericAddress(int a, int inverted, int o) {
        addr = a;
        inv = inverted;
        orient = o;
    }

    public String toLongString() {
        if (addr < LBMIN) {
            return "SX-Adr " + (addr / 10) + "." + (addr % 10);
        } else {
            return "V-Adr " + addr;
        }
    }

    public String toString() {
        if (addr < LBMIN) {
            return (addr / 10) + "." + (addr % 10);
        } else {
            return "" + addr;
        }
    }
}

File: src/main/java/de/blankedv/sx4draw/GenericAddress.kt
/*
SX4Draw
Copyright (C) 2019 Michael Blank

This program is free software: you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation, either version 3 of the License, or
(at your option) any later version.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.

You should have received a copy of the GNU General Public License
along with this program. If not, see <http://www.gnu.org/licenses/>.
 */

package de.blankedv.sx4draw

import de.blankedv.sx4draw.Constants.INVALID_INT
import de.blankedv.sx4draw.Constants.LBMIN

/**
 * @author mblank
 */


data class GenericAddress (var addr :Int, var addr2 :Int = INVALID_INT, var inv : Int = 0, var orient : Int  = 0) {

    override fun toString(): String {
        return if (addr < LBMIN) {
            (addr / 10).toString() + "." + addr % 10 + " inv="+inv + " orient="+orient
        } else {
            "" + addr  + " inv="+inv + " orient="+orient
        }
    }

}
